onnxruntime/core/providers/rocm/rocm_execution_provider_info.cc

```diff
@@ -13,6 +13,8 @@ namespace onnxruntime {
 namespace rocm {
 namespace provider_option_names {
 constexpr const char* kDeviceId = "device_id";
+constexpr const char* kHasUserComputeStream = "has_user_compute_stream";
```

Comment:
We will deprecate "has_user_compute_stream" provider option since it is implicitly set by user_compute_stream.

Reply:
How about use `std::optional<void*>user_compute_stream` or `**user_compute_stream`?
Thus `user_compute_stream=nullptr` wouldn't be ambiguous.

Reply:
Yes, use optional seems good, shell we remove 'has_user_compute_stream' now? or sometimes later? @tianleiwu

Reply:
We can remove 'has_user_compute_stream' later.